scRNA-seq-advanced/03-dataset_integration.Rmd

```diff
@@ -89,45 +93,39 @@ We could also use a `for` loop, which is the approach that many programming lang
 A different and more modular coding approach to reading in these files (and more!) is to leverage the [`purrr`](https://purrr.tidyverse.org/) `tidyverse` package, which provides a convenient set of functions for operating on lists.
 You can read more about these functions and their power and utility in R in [Chapter 21 of _R for Data Science_](https://r4ds.had.co.nz/iteration.html#for-loops-vs.-functionals).
 
-Of particular interest is the [`purrr::map()`](https://purrr.tidyverse.org/reference/map.html) family of functions, which can be used to run a given function on each element of a list or vector in one call.
+Of particular interest is the [`purrr::map()`](https://purrr.tidyverse.org/reference/map.html) family of functions, which can be used to run a given function on each element of a list (or vector) in one call.
 The general syntax for `purrr::map()` and friends is:
 
 ```
 # Syntax for using the map function:
-purrr::map(<input vector or list>, 
-           <function to apply to each item in the input>, 
-           <any additional arguments to the function can go here>, 
-           <and also here if there are even more arguments, and so on>)
+purrr::map(input list or vector, 
+           function to apply to each item in the input, 
+           any additional arguments to the function can go here, 
+           and also here if there are even more arguments, and so on)
 ```
 
 
-Let's see a very simple example in action by quickly taking the square root of a vector of values:
-
-```{r map example, live = TRUE}
-
-# Define a vector of squares
-squares <- c(4, 9, 16)
-
-# get the square root of each number without purrr
-sqrt(squares[1])
-sqrt(squares[2])
-sqrt(squares[3])
-
-# The `purrr::map()` function always returns a list
-purrr::map(squares, # vector to map over
-           sqrt)    # function to apply to each item in `squares`
-```
-
 The output from running `purr::map()` is always a list (but note that there are other `purrr::map()` relatives which return other object types, as you can read about in [the `purrr::map()` documentation](https://purrr.tidyverse.org/reference/index.html)).
+If this concept sounds a little familiar to you, that's because it probably is!
+In fact, this is conceptually the same as base R's `lapply` function, and the `purrr::map()` family of functions provide more robust and reliable alternatives to the base R `apply` functions.
```

Comment:
I'm trying to hedge this a bit... I don't think `purrr` provides a good alternative to `apply()` itself!

```suggestion
Base R's `lapply()` function can provide similar utility, and the `purrr::map()` family of functions can in part be thought of as an alternative to some of the base R `apply` functions, with more consistent behavior.
```